File: back/kotlin/app/delivery/feature/game/endpoints/GameEndpoints.kt
package delivery.feature.game.endpoints

import io.ktor.server.*
import io.ktor.http.*
import io.ktor.server.application.*
import io.ktor.server.plugins.cors.*
import io.ktor.server.response.*
import io.ktor.server.routing.*
import kotlinx.coroutines.*

class GameEndpoints(private val controller: GameController, private val application: Application) {
    
    fun gameRoutes() {

        application.routing {

            options {
//                call.respondPreflight()
            }

            get("/hello") {
                call.respondText("Hello Kotlin!")
            }

            get("/start") {
                val call = call
                val job = launch {
                    controller.startGame(call)
                }
                job.join()
            }

            get("/restart") {
                val call = call
                val job = launch {
                    controller.restartGame(call)
                }
                job.join()
            }

            post("/move") {
                val call = call
                val job = launch {
                    controller.makeMove(call)
                }
                job.join()
            }

        }

    }
    
}

File: back/go/app/delivery/feature/game/endpoints/GameEndpoints.go
package endpoints

import (
	"github.com/gorilla/mux"
	game "go-ttt/app/delivery/feature/game/controllers"
)

func GameEndpoints(r *mux.Router, controller *game.GameController) {
	r.HandleFunc("/hello", controller.HelloGo).Methods("GET")
	r.HandleFunc("/start", controller.StartGame).Methods("GET")
	r.HandleFunc("/restart", controller.RestartGame).Methods("GET")
	r.HandleFunc("/move", controller.MakeMove).Methods("POST")
}